core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/cluster/stopstartds/StopStartDownscaleFlowService.java

```diff
@@ -51,50 +54,60 @@
+        flowMessageService.fireInstanceGroupEventAndLog(stackId, Status.UPDATE_IN_PROGRESS.name(), hostGroupName,
+                CLUSTER_SCALING_STOPSTART_DOWNSCALE_NODE_STOPPING,
+                String.valueOf(decommissionedFqdns.size()), hostGroupName, String.join(",", decommissionedFqdns));
+    }
+
+    public void instancesStopped(long stackId, List<InstanceMetaData> instancesStopped) {
         instancesStopped.stream().forEach(x -> instanceMetaDataService.updateInstanceStatus(x, InstanceStatus.STOPPED));
-        stackUpdater.updateStackStatus(stackId, DetailedStackStatus.AVAILABLE, "Instances: " + instancesStopped.size() + " stopped successfully.");
+        flowMessageService.fireEventAndLog(stackId, UPDATE_IN_PROGRESS.name(), CLUSTER_SCALING_STOPSTART_DOWNSCALE_NODES_STOPPED,
+                String.valueOf(instancesStopped.size()), instancesStopped.stream().map(x -> x.getInstanceId()).collect(Collectors.joining(", ")));
+    }
 
+    public void logInstancesFailedToStop(long stackId, List<CloudVmInstanceStatus> notStoppedInstances) {
+        // TODO CB-14929: CB-15418 This needs to be an orange message (i.e. not success, not failure). Need to figure out how the UI
+        //  processes these and applies icons.
+        flowMessageService.fireEventAndLog(stackId, UPDATE_FAILED.name(), CLUSTER_SCALING_STOPSTART_DOWNSCALE_NODES_NOT_STOPPED,
+                String.valueOf(notStoppedInstances.size()),
+                        notStoppedInstances.stream().map(x -> x.getCloudInstance().getInstanceId()).collect(Collectors.toList()).toString());
+    }
+
+    public void clusterDownscaleFinished(Long stackId, String hostGroupName, List<InstanceMetaData> instancesStopped) {
+        stackUpdater.updateStackStatus(stackId, DetailedStackStatus.AVAILABLE, "Instances: " + instancesStopped.size() + " stopped successfully.");
         flowMessageService.fireEventAndLog(stackId, AVAILABLE.name(), CLUSTER_SCALING_STOPSTART_DOWNSCALE_FINISHED,
-                hostGroupName == null ? "null" : hostGroupName);
+                hostGroupName, String.valueOf(instancesStopped.size()),
+                instancesStopped.stream().map(InstanceMetaData::getDiscoveryFQDN).collect(Collectors.joining(", ")));
     }
 
     public void handleClusterDownscaleFailure(long stackId, Exception errorDetails) {
         LOGGER.info("Error during stopstart downscale flow: " + errorDetails.getMessage(), errorDetails);
         stackUpdater.updateStackStatus(stackId, DetailedStackStatus.DOWNSCALE_BY_STOP_FAILED,
                 String.format("New node(s) (stopstart) could not be removed from the cluster: %s", errorDetails));
-        flowMessageService.fireEventAndLog(stackId, UPDATE_FAILED.name(), CLUSTER_SCALING_FAILED, "removed to", errorDetails.getMessage());
+
+        // TODO CB-14929: Error handling. In case of a failure - should the status of the instances be moved to something like ORCHESTRATOR_FAILED?
```

Comment:
This will be one candidate for the stack syncer look for the differences in cloud provider state and CM state to set the instance status appropriately. Right now it shows as running even if it failed to start or stop.